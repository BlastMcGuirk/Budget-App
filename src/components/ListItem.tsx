import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { FontSizes, Layouts } from '../styles/global';

export interface Item {
    name: string;
    amount: number;
    date: string;
    category?: string;
}

export interface ListItemProps {
    item: Item
}

export default function ListItem(props: ListItemProps) {
    const { name, amount, date, category } = props.item;
    return (
        <View style={styles.container}>
            <View style={Layouts.row}>
                <Text style={FontSizes.L}>{name}</Text>
                <Text style={FontSizes.L}>{amount}</Text>
            </View>
            <View style={Layouts.row}>
                <Text style={FontSizes.S}>{category}</Text>
                <Text style={FontSizes.S}>{date}</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        padding: 16,
        borderBottomColor: 'gray',
        borderBottomWidth: 1
    },
})