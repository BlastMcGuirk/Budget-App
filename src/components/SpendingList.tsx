import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { FontSizes } from '../styles/global';
import ListItem, { Item } from './ListItem';

export interface SpendingListProps {
    budget: string;
    items: Item[];
}

export default function SpendingList(props: SpendingListProps) {
    return (
        <View style={styles.container}>
            <View style={styles.header}>
                <Text style={FontSizes.L}>{props.budget}</Text>
            </View>
            {props.items.map((item, index) => {
                return <ListItem key={item.name + index} item={item}/>
            })}
            <View style={styles.footer}>
                <Text style={FontSizes.S}>New Entry...</Text>
            </View>
        </View>
    )
}

const padding = 8;

const styles = StyleSheet.create({
    container: {
        width: '100%',
        borderTopWidth: 2,
    },
    header: {
        borderBottomWidth: 2,
        borderBottomColor: 'gray',
        padding
    },
    footer: {
        borderBottomWidth: 2,
        padding
    },

});