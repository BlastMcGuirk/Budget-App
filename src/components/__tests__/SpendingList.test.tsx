import renderer from 'react-test-renderer';
import SpendingList from '../SpendingList';
import { Item } from '../ListItem';

describe("Spending List Tests", () => {

    const dummyItems: Item[] = [
        {
            name: "Rent",
            category: "Bills",
            amount: 852.25,
            date: "8/4/2022"
        },
        {
            name: "Groceries",
            amount: 130.44,
            date: "8/7/2022"
        }
    ]

    it("Renders correctly", () => {
        const tree = renderer
            .create(<SpendingList budget='name' items={dummyItems} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});