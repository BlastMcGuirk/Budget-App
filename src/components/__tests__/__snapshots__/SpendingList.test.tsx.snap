// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spending List Tests Renders correctly 1`] = `
<View
  style={
    Object {
      "borderTopWidth": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomWidth": 2,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      name
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Rent
      </Text>
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        852.25
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Bills
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        8/4/2022
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Groceries
      </Text>
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        130.44
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        8/7/2022
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 2,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      New Entry...
    </Text>
  </View>
</View>
`;
